engine/src/operators/GroupBy.cpp

```diff
@@ -649,50 +605,53 @@ void distributed_aggregations_without_groupby(Context & queryContext,
 	static CodeTimer timer;
 	timer.reset();
 
-	std::vector<gdf_column_cpp> aggregatedTable = compute_aggregations(input,
-		group_column_indices,
-		aggregation_types,
-		aggregation_input_expressions,
-		aggregation_column_assigned_aliases);
-
-	Library::Logging::Logger().logInfo(
-		timer.logDuration(queryContext, "distributed_aggregations_without_groupby part 1 compute_aggregations"));
-	timer.reset();
-
-	if(queryContext.isMasterNode(CommunicationData::getInstance().getSelfNode())) {
-		queryContext.incrementQuerySubstep();
-		std::vector<ral::distribution::NodeColumns> partitionsToMerge =
-			ral::distribution::collectPartitions(queryContext);
-		partitionsToMerge.emplace_back(CommunicationData::getInstance().getSelfNode(), aggregatedTable);
-
-		std::vector<int> groupColumnIndices(group_column_indices.size());
-		std::iota(groupColumnIndices.begin(), groupColumnIndices.end(), 0);
-		aggregationsMerger(partitionsToMerge, groupColumnIndices, aggregation_types, input);
-		Library::Logging::Logger().logInfo(timer.logDuration(
-			queryContext, "distributed_aggregations_without_groupby part 2 collectPartitions aggregationsMerger"));
-		timer.reset();
-	} else {
-		std::vector<gdf_column_cpp> empty_output_table(aggregatedTable.size());
-		for(size_t i = 0; i < empty_output_table.size(); i++) {
-			empty_output_table[i].create_empty(aggregatedTable[i].get_gdf_column()->type().id(), aggregatedTable[i].name());
-		}
+	// TODO percy william alex port distribution
```

Comment:
put the single node version in here, have an if statement at the beginning to see if you need to do the distributed pre steps like distributing and gathering